File: Java Spring Boot Project/misc/nport_project/MtCProject1/src/main/java/com/nikosportolos/MtCProject1/models/responses/AllCompaniesResponse.java
package com.nikosportolos.MtCProject1.models.responses;

import com.nikosportolos.MtCProject1.models.Company;

import java.util.List;

public class AllCompaniesResponse {

    private List<CompanyResponse> companies;

    public AllCompaniesResponse(List<CompanyResponse> companies) {
        this.companies = companies;
    }

    public List<CompanyResponse> getCompanies() {
        return companies;
    }

    public void setCompanies(List<CompanyResponse> companies) {
        this.companies = companies;
    }
}

File: Java Spring Boot Project/company_directory_app/lib/models/responses/all_companies_response.dart
import 'package:company_directory_app/models/responses/company_response.dart';

class AllCompaniesResponse {
  final List<CompanyResponse> companies;

  AllCompaniesResponse({this.companies});

  factory AllCompaniesResponse.fromJson(Map<String, dynamic> json) {
    var list = json['companies'] as List;
    List<CompanyResponse> itemsList = list.map((i) => CompanyResponse.fromJson(i)).toList();

    return AllCompaniesResponse(companies: itemsList);
  }
}
